colossus-metrics/src/main/scala/colossus/metrics/Collection.scala

```diff
@@ -142,169 +23,89 @@ trait TickedCollector extends EventCollector {
-   *
-   */
-  def tick(interval: FiniteDuration): MetricMap = {
-    //do not move this import unless you want to have a bad time
-    import scala.collection.JavaConversions._
-    metrics.values.collect{
-      case t: TickedCollector => t.tick(interval)
+  def get(tags: T): Option[Long] = Option(map.get(tags)).map{_.get}
+
+  def snapshot(pruneEmpty: Boolean, reset: Boolean): Map[T, Long] = {
+    val keys = map.keys
+    var build = Map[T, Long]()
+    while (keys.hasMoreElements) {
+      val key = keys.nextElement
+      val value: Long = if (reset) {
+        val v = map.get(key).getAndSet(0L)
+        // notice there's a race condition since between these two lines another
+        // increment could occur and subsequently be lost, but it's an
+        // acceptable loss
+        if (pruneEmpty && v == 0) {
+          map.remove(key, 0L)
+        }
+        v
+      } else {
+        map.get(key).get
+      }
+      build = build + (key -> value)
     }
-    val context = CollectionContext(globalTags, interval)
-    val eventMetrics = metrics.values.map{_.metrics(context)}
-    val producerMetrics = localProducers.map{_.metrics(context)}
-    //TODO: using <+> here should be unnecessary, might want to use ++
-    (eventMetrics ++ producerMetrics)
-      .foldLeft[MetricMap](Map()){case (build, m) => build <+> m}
-      .filter{! _._2.isEmpty}
-  }
 
-  def subCollection(subSpace: MetricAddress = MetricAddress.Root, subTags: TagMap = TagMap.Empty) = {
-    new LocalCollection(subSpace, subTags, intervals, metrics, Some(this))
+    build
   }
 
 }
 
-object LocalCollection {
-  sealed trait EventResult
-  sealed trait EventError extends EventResult
-
-  case object InvalidEvent extends EventError
-  case object UnknownMetric extends EventError
-  case object Ok extends EventResult
-}
-
+class DuplicateMetricException(message: String) extends Exception(message)
 
-/**
- * A Shared collection is a collection where every metric it returns is
- * completely thread-safe.  A Shared collection is useful if you want to pass
- * one collection to many actors, but be aware that since every event is sent
- * as a separate actor message (as opposed to local collections which collect
- * events just as function calls), this should not be used for very-high
- * frequency events.
- */
-class SharedCollection private[colossus](val local: LocalCollection, val collector: ActorRef) extends Collection[SharedLocality] {
+class Collection(val config: CollectorConfig) {
 
-  import Collection.ParamsFor
+  val collectors = new ConcurrentHashMap[MetricAddress, Collector]
 
-  def getOrAdd[T <: EventCollector : ClassTag, P : ParamsFor[T]#Type](params: P)(implicit creator: Generator[T, P]): Shared[T] = {
-    //create the actual collector
-    local.getOrAdd(params)
-    //get a shared interface for it
-    creator.shared(params, CollectorConfig(local.intervals))(collector)
+  def add(collector: Collector) {
+    collectors.put(collector.address, collector)
```

Comment:
Hrm..do you think it would be wise to have a few different flavors of this?  ie: in this impl, you always clobber what is already there.  Does it make sense for a few variations, or flags on this function which control whether or not you evict something if it is already there?  More of a defensive measure than anything, but still a minor point